test/types/community/collection/bulkWrite.test-d.ts

```diff
@@ -0,0 +1,254 @@
+import { expectError } from 'tsd';
+import { MongoClient, ObjectId } from '../../../../src/index';
+
+// collection.bulkWrite tests
+const client = new MongoClient('');
+const db = client.db('test');
+
+interface SubTestSchema {
+  field1: string;
+  field2: string;
+}
+
+type FruitTypes = 'apple' | 'pear';
+
+// test with collection type
+interface TestSchema {
+  _id: ObjectId;
+  stringField: string;
+  numberField: number;
+  optionalNumberField?: number;
+  dateField: Date;
+  fruitTags: string[];
+  maybeFruitTags?: FruitTypes[];
+  readonlyFruitTags: ReadonlyArray<string>;
+  subInterfaceField: SubTestSchema;
+  subInterfaceArray: SubTestSchema[];
+}
+const collectionType = db.collection<TestSchema>('test.update');
+
+const testDocument: TestSchema = {
+  _id: new ObjectId(),
+  stringField: 'foo',
+  numberField: 123,
+  dateField: new Date(),
+  fruitTags: ['apple'],
+  readonlyFruitTags: ['pear'],
+  subInterfaceField: {
+    field1: 'foo',
+    field2: 'bar'
+  },
+  subInterfaceArray: []
+};
+const { _id, ...testDocumentWithoutId } = testDocument;
+
+// insertOne
+
+collectionType.bulkWrite([
+  {
+    insertOne: {
+      document: testDocument
+    }
+  }
+]);
+collectionType.bulkWrite([
+  {
+    insertOne: {
+      document: testDocumentWithoutId
+    }
+  }
+]);
+expectError(
```

Comment:
in all of the expect errors, might be nice to call out why they are expected to error

Reply:
leaving open for ticket